gpMgmt/bin/pythonSrc/subprocess32/_posixsubprocess.c

```diff
@@ -791,6 +791,7 @@ subprocess_cloexec_pipe(PyObject *self, PyObject *noargs)
         if (write_fd < 0)  /* We don't support F_DUPFD_CLOEXEC / other error */
 #endif
         {
+            write_fd = fds[1];
```

Comment:
When F_DUPFD_CLOEXEC is not defined, `write_fd = fds[1]` is executed twice: here and at the 785th line.